### ISSUE TITLE
jr:itext not resolving correctly as param for external app (and possibly elsewhere)

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#5116]
#### Software and hardware versions 
Collect v2022.1

#### Problem description
![device-2022-04-20-115109](https://user-images.githubusercontent.com/967540/164303061-f40337ab-cb40-4e2f-93f2-8e4f009972eb.png)

#### Steps to reproduce the problem
Fill out the second question on [this form](https://github.com/getodk/counter/blob/master/docs/counter.xlsx)

#### Expected behavior
The translated form name and label should be passed on.

#### Other information 
I've reproduced this in v2022.1 so it's not entirely recent.

I haven't tried using `jr:itext` in any other context or looked at what kinds of JavaRosa test coverage there is. That might be my next step.

**Comment by lognaturel** [comment]
This doesn't look like it has to be in the hotfix to me but it would be good to do a little bit more exploration to confirm there isn't a deeper problem.

**Comment by grzesiek2010** [comment]
This is not a bug in Collect. The generated form contained wrong root node names (`data` instead of `counter`) in paths: `jr:itext('/data/counter2:label')` instead of `jr:itext('/counter/counter2:label')`. The problem was caused by not specified name in settings: https://xlsform.org/en/#specify-xforms-root-node-name which I fixed in https://github.com/getodk/counter/pull/21. it worked well in `XLSForm Online v1.x` (based on a file name I guess) but if you use `XLSForm Online v2.x` it is required.
